src/backend/distributed/commands/function.c

```diff
@@ -259,6 +285,55 @@ create_distributed_function(PG_FUNCTION_ARGS)
 }
 
 
+/*
```

Comment:
I find this definition/function name a little confusing.

We only check `idempotency` when the function is not colocated with anything?  I mean, even if we re-create the same colocated/deletegated function, we don't get into this check (which is true):
```SQL
 SELECT create_distributed_function('func_to_colocate(int)', colocate_with:='tbl_to_colocate_ref');
 SELECT create_distributed_function('func_to_colocate(int)', colocate_with:='tbl_to_colocate_ref');
 SELECT create_distributed_function('func_to_colocate(int)', colocate_with:='tbl_to_colocate_ref');
```

So, maybe we call `RecreateSameNonColocatedFunction` or such?